components/script/dom/globalscope.rs

```diff
@@ -3261,6 +3269,25 @@ impl GlobalScope {
             .expect("Failed to queue update gamepad state task.");
     }
 
+    pub fn handle_gamepad_haptic_effect_completed(&self, index: usize) {
```

Comment:
As said in the other comment, let's move this to an IPC route to simplify the message handling flow.